pkg/proxy/proxy.go

```diff
@@ -520,10 +519,7 @@ func (p *Proxy) CreateOrUpdateRedirect(l4 policy.ProxyPolicy, id string, localEn
 				err := p.ackProxyPort(pp)
 				proxyPortsMutex.Unlock()
 				if err != nil {
-					// Finalize functions can't error out. This failure can only
-					// happen if there is an internal Cilium logic error regarding
-					// installation of iptables rules.
-					panic(err)
+					log.WithError(err).Errorf("ackProxyPort failed on %s", pp.name)
```

Comment:
What does `ackProxyPort failed` mean to the user? Can we make this less terse to give the user better clues if this happens, maybe some notion of the impact on policy implementation or something?

Reply:
Changed this to "Datapath proxy redirection cannot be enabled for %s, L7 proxy may be bypassed". The error contains more detail, such as `fmt.Errorf("Cannot install proxy rules for %s: %s", pp.name, err)`